### ISSUE TITLE
Enable cv::VideoCapture to use arbitrary device-names

### ISSUE THREAD

**Issue by opencv-pushbot** [Itseez/opencv#4630]
Transferred from http://code.opencv.org/issues/3288
```
|| Nikolas Engelhard on 2013-09-26 17:28
|| Priority: Low
|| Affected: None
|| Category: highgui-camera
|| Tracker: Feature
|| Difficulty: 
|| PR: 
|| Platform: None / None
```

Enable cv::VideoCapture to use arbitrary device-names
-----------
```
I have defined several udev rules to distinguish several cameras of the same type. I therefore have a softlink /dev/webcam1 to /dev/video1.
VideoCapture only to allows to open the second filename via open(1), but can't handle the softlink. 
It would be great to have a constructor like 

    capture = cv::VideoCapture("(dev/webcam1", CV_CAP_V4L2)
```

History
-------
##### Vladislav Vinogradov on 2013-09-27 07:32
```
Hello Nikolas,

Thank you for your proposal!

Unfortunately our human resource are limited and it would be very appreciated if you could make a pull request with this feature.
Please see http://www.code.opencv.org/projects/opencv/wiki/How_to_contribute for details.
-   Priority changed from Normal to Low
-   Description changed from I have defined several udev rules to
    distinguish several cameras of the same... to I have defined several
    udev rules to distinguish several cameras of the same ... More
-   Assignee set to Nikolas Engelhard
-   Category set to highgui-camera
```
##### Anna Kogan on 2013-10-03 11:14
```
-   Status changed from New to Open
```

**Comment by renu285** [comment]
is this issue still valid in the latest branch?

**Comment by paroj** [comment]
fixed by #6391
